L1Trigger/Phase2L1ParticleFlow/plugins/PFTrackProducerFromL1Tracks.cc

```diff
@@ -0,0 +1,86 @@
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "MagneticField/Engine/interface/MagneticField.h"
+#include "MagneticField/Records/interface/IdealMagneticFieldRecord.h"
+
+#include "DataFormats/L1TParticleFlow/interface/PFTrack.h"
+#include "L1Trigger/Phase2L1ParticleFlow/interface/L1TPFUtils.h"
+#include "L1Trigger/Phase2L1ParticleFlow/interface/ParametricResolution.h"
+
+namespace l1tpf {
+  class PFTrackProducerFromL1Tracks : public edm::stream::EDProducer<> {
+  public:
+    explicit PFTrackProducerFromL1Tracks(const edm::ParameterSet &);
+    ~PFTrackProducerFromL1Tracks() override {}
+
+  private:
+    edm::EDGetTokenT<std::vector<l1t::PFTrack::L1TTTrackType>> TrackTag_;
+    int nParam_;
+    float fBz_;
+    l1tpf::ParametricResolution resolCalo_, resolTrk_;
+
+    void produce(edm::Event &, const edm::EventSetup &) override;
+
+    void beginRun(edm::Run const &, edm::EventSetup const &iSetup) override {
+      edm::ESHandle<MagneticField> magneticField;
+      iSetup.get<IdealMagneticFieldRecord>().get(magneticField);
```

Comment:
Please consider using `ESGetToken`. Using `ESWatcher` in `produce()` is preferred over caching values in `beginRun()` (e.g. will work with any IOV changes).

Reply:
Switched to ESWatcher + ESGetToken